editor_test.go

```diff
@@ -73,17 +79,148 @@ func TestEditorRender(t *testing.T) {
 		},
 	}
 
-	outputBuffer := bytes.NewBufferString("")
-	terminal.Stdout = outputBuffer
-
 	for _, test := range tests {
-		outputBuffer.Reset()
+		r, w, err := os.Pipe()
+		assert.Nil(t, err, test.title)
+
 		test.data.Editor = test.prompt
-		err := test.prompt.Render(
+		err = test.prompt.Render(
+			terminal.Stdio{Out: w},
 			EditorQuestionTemplate,
 			test.data,
 		)
 		assert.Nil(t, err, test.title)
-		assert.Equal(t, test.expected, outputBuffer.String(), test.title)
+
+		w.Close()
+		var buf bytes.Buffer
+		io.Copy(&buf, r)
+
+		assert.Contains(t, buf.String(), test.expected, test.title)
+	}
+}
+
+func TestEditorPrompt(t *testing.T) {
+	tests := []struct {
+		name      string
+		question  Editor
+		procedure func(*expect.Console)
+		expected  string
+	}{
+		{
+			"Test Editor prompt interaction",
+			Editor{
+				Message: "Edit git commit message",
+			},
+			func(c *expect.Console) {
+				c.Expect("Edit git commit message [Enter to launch editor]")
+				c.SendLine("")
+				go c.ExpectEOF()
+				time.Sleep(time.Millisecond)
+				c.Send("iAdd editor prompt tests\x1b")
+				c.SendLine(":wq!")
```

Comment:
The editor tests are failing for me since I have my EDITOR set to emacs.

Reply:
I get a little further with by forcing the EDITOR, but the DSR codes seems to be going to my console, and the tests hang for me after that:
```
$ EDITOR=/usr/bin/vi go test
^[[47;178R^[[31;22R^[[31;23R^[[31;24R^[[31;25R^[[31;26R^[[31;27R^[[31;28R^[[31;29R^[[31;30R^[[31;31R^[[31;32R
<hangs>
```

Not sure if it is a OSX thing or if there is a race condition with the tests.  When I look at the process table, there is no vim subprocess, so it looks like it getting past the editor tests.

Reply:
I narrowed the hanging test to: `TestMultiSelectPrompt/Test_MultiSelect_prompt_interaction_with_vim_mode`

So I suspect vim-mode require some external configuration?  So might not be related to vt10x testing.

Reply:
There was an issue with runes not being fully read via `ExpectEOF`. This was fixed in: https://github.com/Netflix/go-expect/commit/517479316a06a1ce93e1e22bc7f2e7a8f9713dc2